attestation-agent/attester/src/tdx/mod.rs

```diff
@@ -31,14 +32,17 @@ pub struct TdxAttester {}
 #[async_trait::async_trait]
 impl Attester for TdxAttester {
     async fn get_evidence(&self, mut report_data: Vec<u8>) -> Result<String> {
-        if report_data.len() > 64 {
+        if report_data.len() > TDX_REPORT_DATA_SIZE {
             bail!("TDX Attester: Report data must be no more than 64 bytes");
         }
 
-        report_data.resize(64, 0);
+        report_data.resize(TDX_REPORT_DATA_SIZE, 0);
 
         let tdx_report_data = tdx_attest_rs::tdx_report_data_t {
-            d: report_data.as_slice().try_into()?,
+            d: report_data
+                .as_slice()
+                .try_into()
+                .unwrap_or([0; TDX_REPORT_DATA_SIZE]),
```

Comment:
If we need a default value, it's better to have a `warn!` or `info!` to tell this as it is often unexpected to the caller.

BTW, the error might not occur

As the try into from `&[u8]` to [u8; 64] will be performed by `TryFrom<&[T]>` defined in https://doc.rust-lang.org/std/primitive.slice.html#impl-TryFrom%3C%26%5BT%5D%3E-for-%5BT;+N%5D, which will return true if `slice.len() == N.` that is asserted by `.resize()`.

Reply:
> which will return true if `slice.len() == N.` that is asserted by `.resize()`.

I was reading about the same that the error cannot happen and that was one of the reasons for this PR: just to make sure the size is always correctly set. What I also wanted to add but forgot is a comment about it.

What would be your preference? Simply `.unwrap()` after all?